dynamic_programming/parentheses.py

```diff
@@ -0,0 +1,58 @@
+import math
+
+
+def calculate(a, b, op) -> int:
+    """
+    Return the value after evaluating the expression
+    """
+    if op == "+":
+        return a + b
+    elif op == "-":
+        return a - b
+    else:
+        return a * b
+
+
+def MinandMax(M, m, i, j, operator) -> int:

```

Comment:
Single letter variable names are old school.  Python naming conventions say that capital letters are for constants.